src/interpreter/mod.rs

```diff
@@ -516,41 +616,105 @@ fn parse_term(
 
 fn parse_literal(
     token: &Token,
-    writer: &mut VMWriter<FenderTypeSystem>,
+    engine: &mut ExecutionEngine<FenderTypeSystem>,
     scope: &mut LexicalScope,
 ) -> InterpreterResult {
     let token = &token.children[0];
     Ok(match token.get_name().as_deref().unwrap() {
-        "int" => FenderValue::Int(token.get_match().parse()?).into(),
-        "float" => FenderValue::Float(token.get_match().parse()?).into(),
-        "boolean" => FenderValue::Bool(token.get_match().parse()?).into(),
-        "string" => parse_string(token, writer, scope)?,
+        "int" => FenderValue::Int(unwrap_rust!(token.get_match().parse())?).into(),
+        "float" => FenderValue::Float(unwrap_rust!(token.get_match().parse())?).into(),
+        "boolean" => FenderValue::Bool(unwrap_rust!(token.get_match().parse())?).into(),
+        "string" => parse_string(token, engine, scope)?,
         "char" => parse_char(token)?.into(),
-        "list" => parse_list(token, writer, scope)?,
+        "list" => parse_list(token, engine, scope)?,
         "null" => FenderValue::Null.into(),
-        "closure" => parse_closure(token, writer, scope)?,
+        "closure" => parse_closure(token, engine, scope)?,
+        "structInstantiation" => parse_struct_instantiation(token, engine, scope)?,
         name => unreachable!("{name}"),
     })
 }
 
+fn parse_struct_instantiation(
+    token: &Token,
+    engine: &mut ExecutionEngine<FenderTypeSystem>,
+    scope: &mut LexicalScope,
+) -> InterpreterResult {
+    let mut name = String::new();
+    let mut fields = HashMap::new();
+    for sub in token.children.iter() {
+        match sub.get_name().as_deref().unwrap() {
+            "name" => {
+                if !engine
+                    .context
+                    .struct_table
+                    .struct_name_index()
+                    .contains_key(&sub.get_match())
```

Comment:
Do we only allow one struct of each name to exist? We should be careful with this